### ISSUE TITLE
Cannot read property 'split' of null error

### ISSUE THREAD

**Issue by demisx** [tgriesser/knex#1134]
``` sh
$ knex -V
Knex CLI version:  0.9.0
Local Knex version:  0.9.0
```

Running `knex migrate:make test` produces this error:

``` sh
$ knex migrate:make test
/usr/local/lib/node_modules/knex/lib/bin/cli.js:104
      var ext = (argv.x || env.configPath.split('.').pop()).toLowerCase();
                                         ^
TypeError: Cannot read property 'split' of null
    at Command.<anonymous> (/usr/local/lib/node_modules/knex/lib/bin/cli.js:104:42)
    at Command.listener (/usr/local/lib/node_modules/knex/node_modules/commander/index.js:301:8)
    at emitTwo (events.js:87:13)
    at Command.emit (events.js:172:7)
    at Command.parseArgs (/usr/local/lib/node_modules/knex/node_modules/commander/index.js:610:12)
    at Command.parse (/usr/local/lib/node_modules/knex/node_modules/commander/index.js:458:21)
    at Liftoff.invoke (/usr/local/lib/node_modules/knex/lib/bin/cli.js:166:13)
    at Liftoff.launch (/usr/local/lib/node_modules/knex/node_modules/liftoff/index.js:163:8)
    at Object.<anonymous> (/usr/local/lib/node_modules/knex/lib/bin/cli.js:186:5)
    at Module._compile (module.js:398:26)
    at Object.Module._extensions..js (module.js:405:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:430:10)
    at startup (node.js:141:18)
    at node.js:1003:3
```

**Comment by elhigu** [comment]
Can you show your knexfile too? Sounds like the migration path is set wrong there.

**Comment by demisx** [comment]
Here it is:
```js
module.exports = {
  development: {
    client: 'pg',
    connection: {
      database: 'data-api-dev',
      user:     '',
      password: ''
    },
    migrations: {
      tableName: 'knex_migrations'
    },
    debug: true
  },

  test: {
    client: 'pg',
    connection: {
      database: 'data-api-test',
      user:     '',
      password: 'password'
    },
    migrations: {
      tableName: 'knex_migrations'
    }
  },

  staging: {
    client: 'pg',
    connection: {
      database: 'data-api-stg',
      user:     '',
      password: 'password'
    },
    pool: {
      min: 2,
      max: 10
    },
    migrations: {
      tableName: 'knex_migrations'
    }
  },

  production: {
    client: 'pg',
    connection: {
      database: 'data-api-prod',
      user:     '',
      password: 'password'
    },
    pool: {
      min: 2,
      max: 10
    },
    migrations: {
      tableName: 'knex_migrations'
    }
  }
};
```

**Comment by s2t2** [comment]
I was getting this error when running `knex migrate:make create_robots`. It stopped once I specified the location of the knexfile `knex migrate:make create_robots --knexfile db/config.js`.

**Comment by NathanKleekamp** [comment]
Hi @s2t2 , what does the directory path look like in your knexfile? Is it relative to the knexfile you specified with the cli flag? Or is it from the project root where you're running the `knex migrate` command?

I've tried specifying a different knexfile path by running `npm run knex migrate:make test_migration --knexfile path/to/knexfile.js`, but I'm still getting the "Cannot read property 'split' of null" error.

If I leave the knexfile.js in the default location, everything works as expected, fwiw.

Thanks a ton! -Nathan

**Comment by s2t2** [comment]
@nathankleekamp try relative to the project root. See also http://data-creative.info/process-documentation/2016/04/09/node-for-rails-developers-part-6a-express-postgresql-datastore/

**Comment by NathanKleekamp** [comment]
@s2t2 Thanks for your help, and nice write up! Here's what ended up working for me in case it's helpful for others.

I had installed knex locally, not globally, so to run the knex cli commands, I added a knex script to my `package.json` file like so:

```js
"scripts": {
    "knex": "knex"
  }
```

When trying to create the migrations via `npm run knex migrate:make test_migration --knexfile path/to/knexfile.js`, the `--knexfile` flag wasn't being passed to knex cli via npm.

Adding the `--knexfile` flag to the `package.json` script fixed it:

```js
"scripts": {
    "knex": "knex --knexfile 'path/to/knexfile.js'"
  },
```
